include/phoenix/cffi/Texture.h

```diff
@@ -18,7 +18,7 @@ PXC_API PxTexture* pxTexLoad(PxBuffer* buffer);
 PXC_API PxTexture* pxTexLoadFromVdf(PxVdf const* vdf, char const* name);
 PXC_API void pxTexDestroy(PxTexture* tex);
 
-PXC_API void pxTexGetMeta(PxTexture const* tex, uint32_t* format, uint32_t* width, uint32_t* height, uint32_t* mipmapCount, uint32_t* averageColor);
+PXC_API void pxTexGetMeta(PxTexture const* tex, phoenix::texture_format* format, uint32_t* width, uint32_t* height, uint32_t* mipmapCount, uint32_t* averageColor);
```

Comment:
Could you explain in more detail, why this change is needed? This implementation would break the C-API since `phoenix::texture_format` is a C++-definition which is not available in C (and only works here because when compiling the library, the C++-definitions are imported).

To fix this, the enum's definition should be copied from C++ and its naming scheme should be adjusted to fit in with the C-API.

Reply:
Ah right. You created own enum types within the API. Just for me to understand: Is it best practices? I mean in the end it's just an enum value. And behind the C API is C++ phoenix with its enums. What benefit has it to duplicate the enums?

Nevertheless I will add the C-style enums in a minute...

Reply:
In this case, since I'd like the API to actually be usable from C, no C++ files may be included anywhere since C-compilers won't be able to deal with them correctly. Since `phoenix::texture_format` is defined in a C++-file (namely `phoenix/texture.hh`), it can't be used in this header file directly. This is why a separate C-style enum is required.